File: app/src/main/java/ru/fefu/welcomescreen/welcome_activity.java
package ru.fefu.welcomescreen;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import ru.fefu.activitytracker.R;

public class welcome_activity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        Button regButton = findViewById(R.id.regButton_welcome);
        regButton.setOnClickListener(this);

        Button logButton = findViewById(R.id.logButton_welcome);
        logButton.setOnClickListener(this);

    }

    @Override
    public void onClick(View v) {
        Intent i;
        switch (v.getId()) {
            case R.id.regButton_welcome:
                i = new Intent(this, register_activity.class);
                startActivity(i);
                break;
            case R.id.logButton_welcome:
                i = new Intent(this, login_activity.class);
                startActivity(i);
                break;
        }
    }
}

File: app/src/main/java/ru/fefu/welcomescreen/WelcomeActivity.kt
package ru.fefu.welcomescreen

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.content.Context
import android.content.Intent
import ru.fefu.activitytracker.databinding.ActivityWelcomeBinding
import ru.fefu.mainmenu.MainPartActivity

class WelcomeActivity : AppCompatActivity() {

    lateinit var binding: ActivityWelcomeBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityWelcomeBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val sharedPreferences = getSharedPreferences("user", Context.MODE_PRIVATE)
        if (sharedPreferences.getString("token", "") != "") {
            startActivity(Intent(this, MainPartActivity::class.java))
            finish()
        }

        binding.regButton.setOnClickListener {
            startActivity(Intent(this, RegisterActivity::class.java))
        }
        binding.logButton.setOnClickListener {
            startActivity(Intent(this, LoginActivity::class.java))
        }
    }
}